### PR TITLE
add button to MainPlayer to go to Detail view

### PR DESCRIPTION
**Pull request by csicar**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

## Changes
- Added a `<-` button to the top of the `MainVideoPlayer`

## Reason
The reason this button is useful is, that when opening videos via a link (and having selected "Open Video directly"), you can see the comments and related videos. 

## Example
example

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/player/MainVideoPlayer.java [hunk 1]
```diff
@@ -641,6 +648,9 @@ public void onClick(View v) {
             } else if (v.getId() == repeatButton.getId()) {
                 onRepeatClicked();
                 return;
+            } else if (v.getId() == showDetailButton.getId()){
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/player/MainVideoPlayer.java]: A space is missing between ) and {

File: app/src/main/res/layout/activity_main_player.xml [hunk 2]
```diff
@@ -151,49 +151,61 @@
...
-                        android:clickable="true"
-                        android:focusable="true"
-                        tools:ignore="RtlHardcoded"
-                        tools:text="The Video Title LONG very LONG"/>
-
-                    <TextView
-                        android:id="@+id/channelTextView"
-                        android:layout_width="match_parent"
+                        android:layout_marginTop="5dp"
+                        android:background="@drawable/ic_expand_more_white_24dp" />
```
Comment (Stypox) [inline app/src/main/res/layout/activity_main_player.xml]: The space between " and /> should be removed

**Comment by csicar** [comment]
Requested changes have been implemented. Is there something else missing?

**Comment by Stypox** [comment]
Please wait for a maintainer to review your changes. This could take a while since they are often busy in real life and there are more important and complex PRs on the way to be merged.

**Comment by AdKiller** [comment]
Always wanted this feature. thank you very much for implementing it!

File: app/src/main/res/layout/activity_main_player.xml [hunk 3]
```diff
@@ -151,49 +151,66 @@
...
+                            android:marqueeRepeatLimit="marquee_forever"
+                            android:scrollHorizontally="true"
+                            android:singleLine="true"
+                            android:textColor="@android:color/white"
+                            android:textSize="15sp"
+                            android:textStyle="bold"
+                            android:clickable="true"
+                            android:focusable="true"
+                            tools:ignore="RtlHardcoded"
+                            tools:text="The Video Title LONG very LONG"/>
```
Comment (TobiGr) [inline app/src/main/res/layout/activity_main_player.xml]: What the hell is that? Is there any reason, why we put this here before? I guess, this is replaced by the correct title autmatically, but why do we use this as the default one 😂

Comment (csicar) [reply app/src/main/res/layout/activity_main_player.xml]: I think, that `tools:text` is only used by Android Studio to display a value in the layout preview. During runtime, the value will never be used.

**Comment by Phoenix616** [comment]
I feel like making the button look like a back button kinda breaks expectations: Back implies that you actually go back to the previous screen (like with the device back button), not that you always go to the info page.

I feel like a design that clearly shows the info page (e.g. an i in a circle) and a different placement would solve that.

**Comment by Stypox** [comment]
@Phoenix616 you are right, an info button would be better. This expectation-breaking already exists with other back button: #2358. The proposed fix is basically the same as yours: https://github.com/TeamNewPipe/NewPipe/issues/2358#issuecomment-495395844.
@csicar would you mind changing the back button to some type of info button?

**Comment by csicar** [comment]
The bahaviour oft the back butto is how Android UIs are intended to behave (as intended by the ui guidelines):
- the Back Button in the toolbar of an app should navigate hierachically upwards
- the android's system back button navigates chronologically backwards. 


Reference: https://material.io/design/navigation/understanding-navigation.html#reverse-navigation 

I personally perfer to think of the Button as navigating upwards, but I can understand the Argument. Maybe the  down arrow from YouTube's app could be a compromise?

If not, i'll be happy to implement to info button.

**Comment by Stypox** [comment]
For me the back button is ok, then 😉

**Comment by Phoenix616** [comment]
Interesting that they would suggest such a confusing way of handling things. (Duplicate back buttons with different functionality) I feel like the down arrow would be a lot better  and more intuitive. (Especially for people coming from the original app ;))

**Comment by csicar** [comment]
Example of how it looks with the "expand" button:
2019-08-17-214332_398x582_scrot

[patch.txt](https://github.com/TeamNewPipe/NewPipe/files/3512393/patch.txt)

**Comment by csicar** [comment]
@Phoenix616 I guess having very similar symbols for the different actions doesn't help :)

**Comment by TobiGr** [comment]
I am completely fine with the behaviour and icons you used in the example video :+1:

**Comment by Phoenix616** [comment]
I just realised that the downwards arrow causes another usability problem: The arrow at the right side looks the same but it is used to expand the view of available video options instead.

So a backwards arrow like the Android docs suggest might actually be the better option here :S (unless a different icon can e found for either going to the info view or expanding the option view I guess)

**Comment by TobiGr** [comment]
@Stypox is there anything which prevents this PR from being merged?

**Comment by Stypox** [comment]
@TobiGr no, this is ready, I forgot to approve changes at last ;-)

**Comment by Stypox** [comment]
@csicar could you check the bug report from TobiGr?

**Comment by Stypox** [comment]
Closing as this isn't needed anymore after 2907

**Pull request closed without merging**